src/gt4py/next/embedded/function_field.py

```diff
@@ -0,0 +1,203 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import annotations
+
+import dataclasses
+import operator
+from typing import Any, Callable, TypeAlias, TypeGuard
+
+import numpy as np
+
+from gt4py._core import definitions as core_defs
+from gt4py.next import common
+from gt4py.next.common import Infinity
+from gt4py.next.embedded import common as embedded_common, nd_array_field as nd
+from gt4py.next.embedded.nd_array_field import _get_slices_from_domain_slice
+
+
+_EMPTY_DOMAIN = common.Domain((), ())
+
+ConstantFieldValue: TypeAlias = int | float | complex
+
+
+@dataclasses.dataclass(frozen=True)
+class FunctionField:
+    func: Callable
+    domain: common.Domain = _EMPTY_DOMAIN
+    _constant: bool = False
```

Comment:
not sure if we need that...

Reply:
let's discuss possible use-cases...

Reply:
I added this special case since if I just pass `lambda: value` to `np.fromfunction` it fails as it passes the indices to the lambda, so instead I use `np.full` with constant fields.

```python
return np.fromfunction(lambda *indices: self.func(*indices), shape)
E   TypeError: constant_field.<locals>.<lambda>() takes 0 positional arguments but 2 were given
```